EasyPost.Tests/BatchTest.cs

```diff
@@ -83,111 +64,137 @@ public async Task TestAll()
 
             List<Batch> batches = batchCollection.batches;
 
-            Assert.IsTrue(batches.Count <= Fixture.PageSize);
-            Assert.IsNotNull(batchCollection.has_more);
-            foreach (var item in batches)
+            Assert.True(batchCollection.has_more);
+            Assert.True(batches.Count <= Fixture.PageSize);
+            foreach (Batch item in batches)
             {
-                Assert.IsInstanceOfType(item, typeof(Batch));
+                Assert.IsType<Batch>(item);
             }
         }
 
-        [TestMethod]
-        public async Task TestCreateAndBuy()
+        [Fact]
+        [CrudOperations.Update]
+        public async Task TestAddRemoveShipments()
         {
-            _vcr.SetUpTest("create_and_buy");
+            UseVCR("add_remove_shipments");
+
+            Shipment shipment = await Client.Shipment.Create(Fixture.OneCallBuyShipment);
+
+            Batch batch = await Client.Batch.Create();
 
-            Batch batch = await Batch.CreateAndBuy(new Dictionary<string, object>
+            batch = await batch.AddShipments(new List<Shipment>
             {
-                {
-                    "shipments", new List<Dictionary<string, object>>
-                    {
-                        Fixture.OneCallBuyShipment
-                    }
-                }
+                shipment
+            });
+
+            Assert.Equal(1, batch.num_shipments);
+
+            batch = await batch.RemoveShipments(new List<Shipment>
+            {
+                shipment
             });
 
-            Assert.IsInstanceOfType(batch, typeof(Batch));
-            Assert.IsTrue(batch.id.StartsWith("batch_"));
-            Assert.AreEqual(1, batch.num_shipments);
+            Assert.Equal(0, batch.num_shipments);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestBuy()
         {
-            _vcr.SetUpTest("buy");
+            UseVCR("buy");
 
             Batch batch = await CreateOneCallBuyBatch();
 
-            await batch.Buy();
+            batch = await batch.Buy();
 
-            Assert.IsInstanceOfType(batch, typeof(Batch));
-            Assert.AreEqual(1, batch.num_shipments);
+            Assert.IsType<Batch>(batch);
+            Assert.Equal(1, batch.num_shipments);
         }
 
-        [TestMethod]
+        [Fact]
+        [CrudOperations.Update]
         public async Task TestCreateScanForm()
         {
-            _vcr.SetUpTest("create_scan_form");
-
+            UseVCR("create_scan_form");
 
             Batch batch = await CreateOneCallBuyBatch();
 
-            await batch.Buy();
+            if (IsRecording())
+            {
+                Thread.Sleep(10000); // Wait enough time to process
+            }
```

Comment:
Issue: As mentioned previously, this sleep should not be here. There is no reason to wait in-between creating a batch and buying one as our API will respond immediately. This only adds an extra unnecessary 10 seconds timeout when re-recording without providing value.

Sleeps are required between buying a batch and the next operation such as creating a label or scanform. Not before.

Reply:
Resolved.